### PR TITLE
Switch to cmake and fix 1250+ errors - rework link implementation

### PR DESCRIPTION
**Pull request by perrettecl**

### CODE REVIEW

**Review by stark-dev — APPROVED** [review]

File: lib/public_includes/fty_asset_dto.h [hunk 1]
```diff
@@ -273,6 +274,41 @@ class Asset
...
+
+    const std::string&  getEndpointProtocolAttribut(uint8_t index, const std::string & attributName) const;
+
+    void setEndpointProtocol(uint8_t index, const std::string & val);
+    void setEndpointPort(uint8_t index, const std::string & val);
+    void setEndpointSubAddress(uint8_t index, const std::string & val);
+    void setEndpointOperatingStatus(uint8_t index, const std::string & val);
+    void setEndpointErrorMessage(uint8_t index, const std::string & val);
+
+    void setEndpointProtocolAttribut(uint8_t index, const std::string & attributName, const std::string & val);
```
Comment (jeffdegott) [inline lib/public_includes/fty_asset_dto.h:307]: typo? : missing setEndpointProtocolAttribut(e), attribut(e)Name

File: server/src/asset-server.cc [hunk 2]
```diff
@@ -164,12 +170,12 @@ void AssetServer::handleAssetManipulationReq(const messagebus::Message& msg)
 
     // clang-format off
     static std::map<std::string, std::function<void(const messagebus::Message&)>> procMap = {
-        { FTY_ASSET_SUBJECT_CREATE,       [&](const messagebus::Message& msg){ createAsset(msg); } },
-        { FTY_ASSET_SUBJECT_UPDATE,       [&](const messagebus::Message& msg){ updateAsset(msg); } },
-        { FTY_ASSET_SUBJECT_DELETE,       [&](const messagebus::Message& msg){ deleteAsset(msg); } },
-        { FTY_ASSET_SUBJECT_GET,          [&](const messagebus::Message& msg){ getAsset(msg); } },
-        { FTY_ASSET_SUBJECT_GET_BY_UUID,  [&](const messagebus::Message& msg){ getAsset(msg, true); } },
-        { FTY_ASSET_SUBJECT_LIST,         [&](const messagebus::Message& msg){ listAsset(msg); } }
+        { FTY_ASSET_SUBJECT_CREATE,       [&](const messagebus::Message& messsage){ createAsset(messsage); } },
```
Comment (jeffdegott) [inline server/src/asset-server.cc:173]: meSSSage

**Review by jeffdegott — APPROVED** [review]

File: lib/public_includes/fty_asset_dto.h (hunk 1 above)
Comment (stark-dev) [reply lib/public_includes/fty_asset_dto.h]: fixed ;)

File: server/src/asset-server.cc (hunk 2 above)
Comment (stark-dev) [reply server/src/asset-server.cc]: sssolved

**Review by boricj — COMMENTED** [review]
Will take a while to read, preliminary comments so far only.

File: lib/public_includes/fty_asset_dto.h [hunk 3]
```diff
@@ -273,6 +274,41 @@ class Asset
         bool forceUpdatedFalse = false);
     void setLinkedAssets(const std::vector<AssetLink>& assets);
     void setSecondaryID(const std::string& secondaryID);
+    void setFriendlyName(const std::string& friendlyName);
+
+    //Wrapper for addresses => max 256
+    using AddressMap = std::map<uint8_t, std::string>;
+
+    AddressMap          getAddressMap() const;
```
[the remark this answers is not in the document]
Comment (perrettecl) [reply lib/public_includes/fty_asset_dto.h:314]: That's not the idea at all. The idea is to mp our map of asset into something more usable

File: lib/test/test.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
+/*  ========================================================================
```
[the remark this answers is not in the document]
Comment (perrettecl) [reply lib/test/test.cpp:1]: nothing for now

File: lib/public_includes/fty_asset_dto.h [hunk 5]
```diff
@@ -273,6 +274,41 @@ class Asset
...
+
+    const std::string&  getEndpointProtocolAttribute(uint8_t index, const std::string & attributeName) const;
+
+    void setEndpointProtocol(uint8_t index, const std::string & val);
+    void setEndpointPort(uint8_t index, const std::string & val);
+    void setEndpointSubAddress(uint8_t index, const std::string & val);
+    void setEndpointOperatingStatus(uint8_t index, const std::string & val);
+    void setEndpointErrorMessage(uint8_t index, const std::string & val);
+
+    void setEndpointProtocolAttribute(uint8_t index, const std::string & attributeName, const std::string & val);
```
[the remark this answers is not in the document]
Comment (perrettecl) [reply lib/public_includes/fty_asset_dto.h:339]: yes but we cannot today as we do not store endpoint

File: lib/src/fty_asset_dto.cc [hunk 6]
```diff
@@ -281,6 +289,119 @@ void Asset::setSecondaryID(const std::string& secondaryID)
...
+    Asset::ProtocolMap protocols;
+
+    for(uint16_t index = 0; index <= 255; index++ )
+    {
+        const std::string & protocol = getEndpointProtocol(static_cast<uint8_t>(index));
+
+        if(!protocol.empty()){
+            protocols[static_cast<uint8_t>(index)] = protocol;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (perrettecl) [reply lib/src/fty_asset_dto.cc:508]: It has been design like that long time ago. we cannot change it today without rewriting half of the project.

File: lib/src/fty_asset_dto.cc [hunk 7]
```diff
@@ -281,6 +289,119 @@ void Asset::setSecondaryID(const std::string& secondaryID)
...
+
+const std::string& Asset::getAddress(uint8_t index) const
+{
+    return getExtEntry("ip."+std::to_string(index));
+}
+
+void Asset::setAddress(uint8_t index, const std::string & address)
+{
+    setExtEntry("ip."+std::to_string(index), address);
+}
```
[the remark this answers is not in the document]
Comment (perrettecl) [reply lib/src/fty_asset_dto.cc:489]: a user will not provide a list of address, because in most of the case he want's to change only one

**Pull request merged**